internal/pkg/types/channel_id.go

```diff
@@ -40,6 +40,18 @@ func (z ChannelID) MarshalJSON() ([]byte, error) {
 	return json.Marshal(z.val)
 }
 
+// MarshalBinary converts a ChannelID to bytes
```

Comment:
the idea was for this structures to implement `Encode/Decode` which can be then use with multiple encoders.